### ISSUE TITLE
KoreBuild is obsolete

### ISSUE THREAD

**Issue by martincostello** [aspnet-contrib/AspNet.Security.OAuth.Providers#321]
As noted here https://github.com/aspnet/BuildTools/pull/977, KoreBuild is now obsolete.

This breaks the #280 branch from 3.0 preview 6 and onwards as there is no version available to download that supports these SDK versions.

The question is - what's the replacement?

ASP.NET Core uses "Arcade" now, but I imagine that's not designed for our needs and (I imagine) not simple to integrate.

To unblock #280 for now, I've just pulled in the simple scripts I use for my own projects until there's an agreed replacement.

/cc @PinpointTownes @poke

**Comment by kevinchalet** [comment]
OpenIddict was updated to use Arcade: as you said, it was designed for Microsoft's needs, so it's super opinionated. Disabling certain features (like packages signing) is a painful process, and you have to dig into the massive documentation to learn how to do that.

That said, I'm convinced it's the right way to go: while it's painful to configure, it works like a charm once it's set up, and it's something we won't have to maintain in the long term.

It's also a cool way to find things that could be simplified or improved. For instance, it will encourage you to remove the `xunit`/`Microsoft.NET.Test.Sdk` references from the tests projects, as they are now part of the SDK (which is something VS doesn't tell you if you don't use Arcade).

**Comment by martincostello** [comment]
I've got no problem with adopting Arcade - I just found the docs a bit impenetrable at first glance, so figured it was worth discussion prior to trying to adopt it and then finding it's not what we wanted to do.

Do you have a link the to PR for converting OpenIddict? I can use that as a basis to migrate the 3.0 branch over before preview 9 lands.

**Comment by kevinchalet** [comment]
> I've got no problem with adopting Arcade - I just found the docs a bit impenetrable at first glance, so figured it was worth discussion prior to trying to adopt it and then finding it's not what we wanted to do.

Makes total sense 😄 

Here are the relevant parts:
  - https://github.com/openiddict/openiddict-core/tree/dev/eng
  - https://github.com/openiddict/openiddict-core/blob/dev/Directory.Build.props
  - https://github.com/openiddict/openiddict-core/blob/dev/Directory.Build.targets
  - https://github.com/openiddict/openiddict-core/blob/dev/global.json
  - https://github.com/openiddict/openiddict-core/blob/dev/Build.cmd
  - https://github.com/openiddict/openiddict-core/blob/dev/build.sh

There's also this very nice sample, that was super helpful: https://github.com/dotnet/arcade-minimalci-sample

**Comment by martincostello** [comment]
Great, thanks Kévin, I'll take a look at that later! 😃

**Comment by kevinchalet** [comment]
Actually, I still have one thing to figure out with Arcade: how to make build versioning work with AppVeyor (that uses incremental identifiers that don't reset every day).

https://github.com/openiddict/openiddict-core/issues/767

**Comment by martincostello** [comment]
OK, cool.

That reminds me - can you add me to the AppVeyor and/or Travis projects for aspnet-contrib? If builds fail randomly I don't have the permissions to press the retry button!

**Comment by kevinchalet** [comment]
For AppVeyor: done (lemme know if it doesn't work).
For Travis: not sure how to do that. I thought GitHub team members were automatically able to restart builds if necessary 😕 

![image](https://user-images.githubusercontent.com/6998306/63214053-d2ff5900-c113-11e9-9ea4-f0b478888a57.png)

**Comment by martincostello** [comment]
Maybe it did work in Travis already and I’ve forgotten! I’ll check again later to make sure.

**Comment by martincostello** [comment]
@PinpointTownes Was just looking at your commit to add Arcade and noticed this boilerplate that's needs updating: https://github.com/openiddict/openiddict-core/blame/dev/LICENSE.md#L190

**Comment by kevinchalet** [comment]
Actually, you can't update this file (and it makes sense when you think about it: this paragraph just refers to how you're supposed to reference the Apache license in the source code files)

If you change it, Arcade will return a nice error:

```
error : License file content '[path]\LICENSE.md' doesn't match the expected license '[path]\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19405.1\tools\Licenses\Apache-2.0.txt'
```

**Comment by martincostello** [comment]
TIL!

**Comment by martincostello** [comment]
Resolved by #335.
